src/verifier.cc

```diff
@@ -15,15 +15,143 @@
 #include "src/verifier.h"
 
 #include <cmath>
+#include <vector>
+
+#include "src/command.h"
 
 namespace amber {
 namespace {
 
-const float kEpsilon = 0.002f;
+const double kEpsilon = 0.002;
 
-bool IsFloatPixelEqualInt(float pixel, uint8_t expected) {
+bool IsFuzzyEqual(double real, double expected) {
   // TODO(jaebaek): Change kEpsilon to tolerance.
-  return std::fabs(pixel - static_cast<float>(expected) / 255.0f) < kEpsilon;
+  return std::fabs(real - expected) < kEpsilon;
+}
+
+template <typename T>
+Result CheckIntValue(ProbeSSBOCommand::Comparator comp,
+                     const uint8_t* memory,
+                     const std::vector<Value>& values) {
+  const T* ptr = reinterpret_cast<const T*>(memory);
+  switch (comp) {
+    case ProbeSSBOCommand::Comparator::kEqual:
+      for (auto v : values) {
+        if (*ptr != static_cast<T>(v.AsUint64()))
+          return Result("Verifier::ProbeSSBO failed");
+        ++ptr;
+      }
+      break;
+    case ProbeSSBOCommand::Comparator::kNotEqual:
+      for (auto v : values) {
+        if (*ptr == static_cast<T>(v.AsUint64()))
+          return Result("Verifier::ProbeSSBO failed");
```

Comment:
These results should have more context:

Result("Verifier failed: " + std::to_string(*ptr) + " != " + std::to_string(val));